Cargo.toml

```diff
@@ -19,6 +19,7 @@ edition = "2018"
 
 [dependencies]
 jobserver = { version = "0.1.16", optional = true }
+os_pipe = "1"
```

Comment:
The msrv CI for windows failed because it automatically pulls in the latest version of `os_pipe` v1.1.2, which depends on windows-sys v0.42.

windows-sys v0.42 seems to use `if`, `match`, `||` and `&&` in `const`, hence failed in rust v1.34.

However, we only need `os_pipe` v1.0.0 to work here, I've tested this with `cargo update --Zminimal-version` and all the tests ran just fine.

So I think this does not break the msrv given that the users manage the `Cargo.lock` to not use the latest version of `os_pipe`, but not entirely sure about that.

Reply:
I agree this doesn't practically update the MSRV. I think a reasonable solution would be to run `cargo update -p os_pipe --precise 1.0.0` (possibly after `cargo generate-lockfile` if needed) in the CI script.

That said, do you know what version these features require? I don't mind bumping it a bit.

Reply:
> That said, do you know what version these features require? I don't mind bumping it a bit.

I looked into the tracking issue https://github.com/rust-lang/rust/pull/57563 and it seems to be the first item in the list https://github.com/rust-lang/rust/issues/49146 , which is stablised in 1.46.

P.S. If we are bumping msrv to 1.46, then I can replace the `Option` used in `fn jobserver()` with `mem::MaybeUninit`.

Reply:
1.46 should be fine. I'll ask if anybody objects on Zulip, but my main goal was to avoid going past Debian stable (which is on 1.48), since it tends to cause a lot of spurious bug reports. Even then, those would probably be directed at the crate that causes the compilation error (in this case windows-sys), rather than us.

That said, I'm slightly unsure about adding the dependency on os_pipe, since it's not nearly as widely used as cc, and is pretty lightweight (so it might be better to implement manually). That said, it might be fine.

Reply:
> (so it might be better to implement manually).

That's definitely doable, though I personally don't want more `unsafe` code in `cc`.

Reply:
Sure, although I think that might be easier unsafe to audit than the current code.

Reply:
It's also notable that we don't currently have any non-optional dependencies, and even among optional dependencies we don't have any that aren't maintained by rust-lang team members (this would change that). So I'm not sure, it might be worth being a bit conservative here.

I'm largely inclined to think windows-sys and libc are fine, though.

Reply:
> So I'm not sure, it might be worth being a bit conservative here.

Hmmm, maybe a good idea to move `os_pipe` into rust-lang considering that it is also a fundamental lib used by other crates?

CC @oconnor663 owner of `os_pipe`.

Reply:
In any case I still need to review this code and use of the dependency (sadly, it's looking like I might not get to this until the weekend, not sure).

Reply:
Happy to help if I can. Luckily the whole crate isn't very much code :)